client.go

```diff
@@ -13,10 +13,11 @@ import (
 )
 
 const (
-	DISCONNECTED = iota
-	DIALING
-	CONNECTING
-	CONNECTED
+	DISCONNECTING 	= 0
```

Comment:
@JulianKropp Can you add back the DIALING state and change the enums back to using `= iota` rather than explicitly numbering the constants?